Marlin/Marlin_main.cpp

```diff
@@ -2419,8 +2425,13 @@ static void clean_up_after_endstop_or_probe_move() {
 
     #elif ENABLED(AUTO_BED_LEVELING_UBL)
 
+      #if ENABLED(UBL_DELTA)
+        if (( ubl.state.active ) && ( ! enable )) {   // leveling from on to off
+          planner.unapply_leveling(current_position);
```

Comment:
There was previously no unapply_leveling for UBL at all, because UBL did not hook the planner.apply_leveling and unapply_leveling routines.  Under UBL_DELTA it does hook these, and I suggest this be enabled for non-delta cartesian UBL as well so that G2/G3 arc, and other blocking_move_to/set_position type of moves will result in leveling on UBL without UBL_DELTA.

I was under the assumption that unapply_leveling was only used when leveling was being disabled, thus returning current_position to what it would have been unleveled.  I will review this logic before commenting further.